### PR TITLE
[Benchmark] Change BigQuery schema

### PR DESCRIPTION
**Pull request by SaveTheRbtz**
This diff changes the BigQuery schema for the `benchmark/cmd/ci` binary so that we can have both long-term perf analysis and adhoc perf tests.

Sadly, this diff is quite enormous and lacks test coverage.  On the bright side it does move core logic out of `main.go` and compartmentalizes the codebase.

[Long-term perf analysis example](https://dapperlabs.grafana.net/d/BLSHi8ZVz/flow-performance-benchmarks?orgId=1&from=now-90d&to=now&var-branch=All&var-env=All&var-type=All):
long term

[Per-test in-depth analysis](https://dapperlabs.grafana.net/d/mA42tHK4z/flow-performance-benchmarks-raw?orgId=1&from=now-24h&to=now&var-branch=All&var-env=All&var-type=All):
short term

Filters for branch, environment, and test type are available:
Filters

Also all tests have annotations to point to a specific commit tested and its branch/tag if available:
Annotations

ref: #3548

### CODE REVIEW

File: integration/benchmark/cmd/ci/main.go [hunk 1]
```diff
@@ -198,279 +180,70 @@ func main() {
...
-		return tps
+	log.Info().Msg("Uploading data to BigQuery")
+	err = db.saveRawResults(
+		ctx,
+		*bigQueryDatasetFlag,
+		*bigQueryRawTableFlag,
+		recorder.BenchmarkResults,
+		*repoInfo,
+		BenchmarkInfo{BenchmarkType: loadType},
+		defaultEnvironment(),
```
Comment (gomisha) [inline integration/benchmark/cmd/ci/main.go:244]: Should the environment be settable? Do we need to add support for other environments to pass to `main` as a flag?

Comment (SaveTheRbtz) [reply integration/benchmark/cmd/ci/main.go:244]: for now, environment is the property of the linux host (kernel version and hostname). -- we'll eventually need to pass a custom one, but that can be done later.

File: integration/benchmark/cmd/ci/main.go [hunk 2]
```diff
@@ -75,42 +53,44 @@ func main() {
...
 	if err != nil {
-		log.Fatal().Err(err).Msg("invalid log level")
+		log.Fatal().Err(err).Str("strLevel", *logLvl).Msg("invalid log level")
 	}
 	log = log.Level(lvl)
 
-	server := metrics.NewServer(log, metricport)
-	<-server.Ready()
-	loaderMetrics := metrics.NewLoaderCollector()
+	git, err := NewGit(log, *gitRepoPathFlag, *gitRepoURLFlag)
```
Comment (pattyshack) [inline integration/benchmark/cmd/ci/main.go:77]: Should repoInfo be optional?  maybe just use a default repoInfo instead of error-ing out

**Review by gomisha — APPROVED** [review]

File: integration/benchmark/cmd/ci/environment.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	// Short description of the environment.
+	EnvironmentName string
+	// GoVersion is the version of the Go runtime.
+	EnvironmentGoVersion string
+	// KernelVersion is the version of the kernel.
+	EnvironmentKernelVersion string
+}
+
+func defaultEnvironment() Environment {
+	kernelFull, _ := io.ReadFile("/proc/version")
```
Comment (pattyshack) [inline integration/benchmark/cmd/ci/environment.go:21]: panic on error?

File: integration/benchmark/cmd/ci/environment.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	kernelFields := strings.Fields(string(kernelFull))
+
+	var kernelVersion = ""
+	if len(kernelFields) >= 3 {
+		kernelVersion = kernelFields[2]
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
```
Comment (pattyshack) [inline integration/benchmark/cmd/ci/environment.go:30]: log error?

File: integration/benchmark/cmd/ci/adjuster.go [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+		timestamp: nowTs,
+		tps:       currentTPS,
+		targetTPS: boundedTPS,
+
+		timedout: uint(currentStats.TxsTimedout),
+		executed: uint(currentStats.TxsExecuted),
+	}, nil
+}
+
+func computeTPS(
```
[the remark this answers is not in the document]
Comment (SaveTheRbtz) [reply integration/benchmark/cmd/ci/adjuster.go:176]: fair!  Will followup with a diff here.  This PR is insanely large at this point.

**Comment by SaveTheRbtz** [comment]
bors merge

**Comment by SaveTheRbtz** [comment]
bors mere

**Pull request merged**